convert2rhel/subscription.py

```diff
@@ -240,14 +240,29 @@ def register_system():
         registration_cmd = RegistrationCommand.from_tool_opts(tool_opts)
 
         try:
-            # The file /etc/os-release is needed for subscribing the system and is being removed with
-            # <system-name>-release package in one of the steps before
-            # RHELC-16
-            os_release_file.restore(rollback=False)
+            # Only in first attempt, in the next attempts the file is already restored from the first run
+            if attempt <= 0:
+                # The file /etc/os-release is needed for subscribing the system and is being removed with
+                # <system-name>-release package in one of the steps before
+                # RHELC-16
+                os_release_file.restore(rollback=False)
+        except (OSError, IOError) as e:
+            loggerinst.critical_no_exit(
+                "Failed to restore the /etc/os-release file needed for subscribing the system with message: %s" % str(e)
+            )
+            raise exceptions.CriticalError(
+                id_="FAILED_TO_SUBSCRIBE_SYSTEM",
+                title="Failed to subscribe system.",
+                description="Failed to restore the /etc/os-release file needed for subscribing the system.",
+                diagnosis="The restore failed with error %s." % (str(e)),
+            )
```

Comment:
Maybe we should use a different id and tittle? We are not subscribing the system at this point, only trying to restore the os_release file. 

The registration comes in the line 261

Reply:
I mean, the procedure is for registering the system, but at this specific step we are not actually registering

Reply:
I was thinking about that, and it resulted in this. I see it that we are trying to subscribe the system and this is part of the process. The problem is then specified in the description and diagnostic. 

But if we would find a better name, I will change that, no problem. I'm not sure which, I felt like `FAILED_TO_RESTORE_OSRELEASE` is a bit misleading.... But I'm not sure there, the naming is the hardest part sometimes :smile:

Reply:
This is a big question and I feel like we need a wiki page or something with all of these things, when to use what, and how each is triggered etc. At the moment it's just "this feels good" but it can lead to confusing things for the customer

Since this prevents the system from being registering it still fails to register, so I feel like this error is still accurate